vyos/vyos-1x
# sla: T4222: Add OWAMP and  TWAMP for service sla

## Change Summary

OWAMP (https://tools.ietf.org/html/rfc4656 ) is a command-line client application and a policy daemon used to determine one-way latencies between hosts.
OWAMP session control uses traditional client-server communication between a control-client and a server,
TWAMP (two-way active measurement protocol) https://tools.ietf.org/html/rfc5357

Add configuration and operation modes


## Types of changes


## Related PR
Required additional pkgs (task in vyos-build)
 owamp-client_4.4.4-1_amd64.deb
 owamp-server_4.4.4-1_amd64.deb
 twamp-client_4.4.4-1_amd64.deb
 twamp-server_4.4.4-1_amd64.deb

https://github.com/vyos/vyos-build/pull/239


## Related Task(s)

* https://phabricator.vyos.net/T4222

## Component(s) name

owamp, twamp
## Proposed changes


## How to test
VyOS sever configuration, set one or both sla servers
```
set service sla owamp-server
set service sla twamp-server
```
From the client site ping a server
```
force owping 192.0.2.120
force twping 192.0.2.190
```

Check services/pids:
```
vyos@r14# ps ax | grep amp
   5049 ?        Ss     0:00 /usr/sbin/owampd -c /etc/owamp-server -R /var/run
   5070 ?        Ss     0:00 /usr/sbin/twampd -c /etc/twamp-server -R /var/run
```

```
vyos@r14:~$ force twping 192.168.122.12
Approximately 12.9 seconds until results available

Directional delays may be inaccurate due to out of sync clocks!

--- twping statistics from [192.168.122.190]:9718 to [192.168.122.12]:9718 ---
SID:	c0a87a0ce64217c616e87d2c0e0e1053
first:	2022-06-01T19:52:23.321
last:	2022-06-01T19:52:33.788
100 sent, 0 lost (0.000%), 0 send duplicates, 0 reflect duplicates
round-trip time min/median/max = 0.446/0.7/0.86 ms, (err=1.38 ms)
send time min/median/max = -294/-293/-293 ms, (err=0.69 ms)
reflect time min/median/max = 294/294/294 ms, (err=0.69 ms)
reflector processing time min/max = 0.00238/0.00811 ms
two-way jitter = 0.2 ms (P95-P50)
send jitter = 0.1 ms (P95-P50)
reflect jitter = 0.1 ms (P95-P50)
send hops = 0 (consistently)
reflect hops = 0 (consistently)
```

## Checklist:
